heka/sandbox/decoders/extract_telemetry_dimensions.lua

```diff
@@ -81,6 +82,15 @@ local function sample(id, sampleRange)
     return hash.crc32(id) % sampleRange
 end
 
+function parse_creation_date(date)
+   local t = dt.rfc3339:match(date or "")
```

Comment:
This will suffice:
```lua
if type(date) ~= "string" then return nil end
```
However, creationDate is not optional so we should failed the message if it doesn't exist or fails to parse.